src/main/src/test/java/org/geoserver/catalog/impl/CatalogImplTest.java

```diff
@@ -2174,9 +2184,109 @@ public void testGetLayerGroupByName() {
         assertNull(catalog.getLayerGroupByName("cite", "layerGroup2"));
     }
 
+    @Test
+    public void testRenameResourceRenamesAssociatedDataRules() throws IOException {
+        // Testing only with a FeatureType because I know SecuredResourceNameChangeListener works at
+        // the ResourceInfo level. If for any reason you change that and don't update the tests
```

Comment:
Non useful comment, nobody will remember it when changing other code. If there is something people need to remember, best put it where they might do the change that would break this assumption instead.

Reply:
removed.